### PR TITLE
🧹 update and clean linux security policy

### PR DESCRIPTION
**Pull request by atomic111**

### CODE REVIEW

File: core/mondoo-linux-security.mql.yaml [hunk 1]
```diff
@@ -388,7 +388,7 @@ queries:
     title: Ensure X Window System is not installed
     impact: 100
     mql: |
-      package("xserver-xorg").installed == false
+      packages.where( name == /^xserver-xorg.*/ ).length == 0
```
Comment (chris-rock) [inline core/mondoo-linux-security.mql.yaml:391]: I recommend:

```
packages.none( name == /^xserver-xorg.*/ )
```

File: core/mondoo-linux-security.mql.yaml [hunk 2]
```diff
@@ -623,15 +623,15 @@ queries:
...
-      if( package("postfix").installed ) {
+      if( package("postfix").installed && service('postfix').running ) {
         parse.ini("/etc/postfix/main.cf").params["inet_interfaces"] == "localhost" || parse.ini("/etc/postfix/main.cf").params["inet_interfaces"] == "loopback-only"
       }
-      if( package("exim4").installed ) {
+      if( package("exim4").installed && service('exim4').running ) {
         parse.ini("/etc/exim4/update-exim4.conf.conf").params["dc_local_interfaces"] == "'127.0.0.1 ; ::1'"
       }
-      ports.listening.all( port != 25 )
+      ports.listening.none(port == 25)
```
Comment (chris-rock) [inline core/mondoo-linux-security.mql.yaml:634]: nice, much easier to read

File: core/mondoo-linux-security.mql.yaml [hunk 3]
```diff
@@ -1080,19 +1081,16 @@ queries:
         ```
   - uid: mondoo-linux-security-auditing-for-processes-that-start-prior-to-auditd-is-enabled
     title: Ensure auditing for processes that start prior to auditd is enabled
-    impact: 80
+    impact: 50
     mql: |
-      if( file("/boot/grub2/grub.cfg" ).exists) {
-        file("/boot/grub2/grub.cfg").content.lines.where( _ == /^[^#]/ ).any(_ == /audit(\s+)?\=(\s+)?1/)
+      if(file("/boot/grub2/grub.cfg").exists) {
+        file("/boot/grub2/grub.cfg").content.contains("audit=1")
```
Comment (chris-rock) [inline core/mondoo-linux-security.mql.yaml:1087]: maybe we can use something like:

```
file("/boot/grub2/grub.cfg").content == /audit\s=\s1/
```

Comment (chris-rock) [reply core/mondoo-linux-security.mql.yaml:1087]: Is it intentional that we do not ignore the comments anymore?

File: core/mondoo-linux-security.mql.yaml [hunk 4]
```diff
@@ -1123,12 +1121,9 @@ queries:
         ```
   - uid: mondoo-linux-security-audit-log-storage-size-is-configured
     title: Ensure audit log storage size is configured
-    impact: 80
+    impact: 40
     mql: |
-      file("/etc/audit/auditd.conf").exists
```
Comment (chris-rock) [inline core/mondoo-linux-security.mql.yaml:1128]: We could also make this a filter of the query itself so the check only runs if the config is there. On the other hand this policy already requires `auditd` to be installed, so the config must exist anyway.

File: core/mondoo-linux-security.mql.yaml [hunk 5]
```diff
@@ -1185,13 +1174,16 @@ queries:
         ```
   - uid: mondoo-linux-security-changes-to-system-administration-scope-sudoers-is-collected
     title: Ensure changes to system administration scope (sudoers) is collected
-    impact: 80
+    impact: 50
+    props:
+      - uid: auditFiles
+        title: Return the content from all /etc/audit/rules.d and /etc/audit/audit.rules
+        mql: |
+          AuditFiles = files.find(from: "/etc/audit/rules.d",regex:'.*\.rules$' , type: "file").list.map(path) + ["/etc/audit/audit.rules"]
```
Comment (chris-rock) [inline core/mondoo-linux-security.mql.yaml:1182]: nit: from readability I find snakeCase with lower case at the beginning easier to read and would be more consistent with the rest of the policy

File: core/mondoo-linux-security.mql.yaml [hunk 6]
```diff
@@ -1329,20 +1322,19 @@ queries:
...
-          ||file('/etc/audit/audit.rules').content.lines.contains(/^(\s+)?\-a\s+always\,exit\s+\-F\s+arch\=b32\s+\-S\s+stime\,settimeofday\,adjtimex\s+\-F\s+key\=time\-change(\s+)?$/)
-        file('/etc/audit/audit.rules').content.lines.contains(/^(\s+)?\-a\s+always\,exit\s+\-F\s+arch\=b64\s+\-S\s+clock\_settime\s+\-k\s+time\-change(\s+)?$/)
-          || file('/etc/audit/audit.rules').content.lines.contains(/^(\s+)?\-a\s+always\,exit\s+\-F\s+arch\=b64\s+\-S\s+clock\_settime\s+\-F\s+key\=time\-change(\s+)?$/)
-        file('/etc/audit/audit.rules').content.lines.contains(/^(\s+)?\-a\s+always\,exit\s+\-F\s+arch\=b32\s+\-S\s+clock\_settime\s+\-k\s+time\-change(\s+)?$/)
-          || file('/etc/audit/audit.rules').content.lines.contains(/^(\s+)?\-a\s+always\,exit\s+\-F\s+arch\=b32\s+\-S\s+clock\_settime\s+\-F\s+key\=time\-change(\s+)?$/)
-        file('/etc/audit/audit.rules').content.lines.contains(/^(\s+)?\-w\s+\/etc\/localtime\s+\-p\s+wa\s+\-k\s+time\-change/)
-      }
+      props.auditFiles.any(_.contains(/^(\s+)?\-a\s+always\,exit\s+\-F\s+arch\=b64\s+\-S\s+adjtimex\s+\-S\s+settimeofday\s+\-k\s+time-change(\s+)?$/))
```
Comment (chris-rock) [inline core/mondoo-linux-security.mql.yaml:1333]: nice simplification of the regex

File: core/mondoo-linux-security.mql.yaml [hunk 7]
```diff
@@ -2913,34 +2908,21 @@ queries:
...
-            name
-            name == props.MondooSudoGroup
-          }
-        }
-      } else {
-        groups.where( name == "sudo" ).list { members {
-            name
-            name == props.MondooSudoGroup
-          }
+      groups.where( name == "wheel" || name == "sudo" ).list != []
```
Comment (chris-rock) [inline core/mondoo-linux-security.mql.yaml:2919]: Maybe we use `groups.any` instead?

```
cnspec> groups.where( name == "wheel" || name == "sudo" ).list != []
[ok] value: [
  0: group id = group/0/wheel
]

cnspec> groups.any( name == "wheel" || name == "sudo" )
[ok] value: true
```

**Review by chris-rock — COMMENTED** [review]
Great improvements @atomic111 I added a some feedback for improvements.

File: core/mondoo-linux-security.mql.yaml (hunk 1 above)
Comment (atomic111) [reply core/mondoo-linux-security.mql.yaml:391]: thanks for the hint

File: core/mondoo-linux-security.mql.yaml (hunk 3 above)
Comment (atomic111) [reply core/mondoo-linux-security.mql.yaml:1087]: sorry my mistake

File: core/mondoo-linux-security.mql.yaml (hunk 5 above)
Comment (atomic111) [reply core/mondoo-linux-security.mql.yaml:1182]: yes, you are right

Comment (atomic111) [reply core/mondoo-linux-security.mql.yaml:1182]: done

File: core/mondoo-linux-security.mql.yaml (hunk 3 above)
Comment (atomic111) [reply core/mondoo-linux-security.mql.yaml:1087]: no, thanks for the hint

File: core/mondoo-linux-security.mql.yaml (hunk 7 above)
Comment (atomic111) [reply core/mondoo-linux-security.mql.yaml:2919]: like it!!!

File: core/mondoo-linux-security.mql.yaml (hunk 4 above)
Comment (atomic111) [reply core/mondoo-linux-security.mql.yaml:1128]: i have introduced two groups to solve it

**Review by chris-rock — APPROVED** [review]
So much more easy to read! Thank you @atomic111

**Pull request merged**